hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -201,6 +209,7 @@ public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
         .withPartition(operation.getPartitionPath())
         .withOptimizedLogBlocksScan(executionHelper.enableOptimizedLogBlockScan(config))
         .withRecordMerger(config.getRecordMerger())
+        .withInstantRange(instantRange)
```

Comment:
Do we need this for all `HoodieMergedLogRecordScanner` builders?  Have you revisited all such scanners?

Reply:
Yeah, for MDT, we have 2 use cases to use the merged log reader: 1. the snapshot reader used by data set table to supply the metadata/stats, 2. the MDT compaction